server/app/services/OffsetBasedPaginationSpec.java

```diff
@@ -0,0 +1,37 @@
+package services;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+import java.util.Optional;
+
+/**
+ * Specifies pagination behavior for a query with offset identifier-based pagination. An offset
+ * identifier identifies the last item of the previous page of results using its sort order
+ * attribute.
+ */
+public class OffsetBasedPaginationSpec<T> {
```

Comment:
any other type that could be used as a sort order. e.g. if we sorted accounts by authority_id we'd need a string identifier.

Reply:
Ah I see. Optional, but maybe would be helpful to add a short comment with examples. I was assuming it would always be a number (but maybe that's just me :))